### DISCUSSION TITLE
Generalize ForecastingGridSearchCV to take multiple models

### DISCUSSION THREAD

**Discussion by arnaujc91** [alan-turing-institute/sktime#5619] [Ideas]
I think a common use case is precisely given some data, how to find the best models with the best hyperparameters. So i was wondering if we could generalize the [ForecastingGridSearchCV](https://www.sktime.net/en/v0.24.1/api_reference/auto_generated/sktime.forecasting.model_selection.ForecastingGridSearchCV.html#forecastinggridsearchcv) object, to take a list of models and the `param_grid` could be a dict with keys as model names and values as grids for those models.

Basically now this takes just one model and finds the best hyper parameters. Next step would be to create a report for all models fed and their hyperparameters.

**Comment by fkiraly** [comment]
this is already possible, you have to use the `MultiplexForecaster`, see its docstring the section "autoML" in the forecasting tutorial for a full vignette.

**Reply by fkiraly** [reply]
(not immediately obvious, but having "model switching" as a separate compositor enables you to build more flexibly, and avoids "uber-merged" estimators)
